Runtimes/Supplemental/cmake/modules/FindSwiftOverlay.cmake

```
@@ -0,0 +1,236 @@
+#[=======================================================================[.rst:
+FindSwiftOverlay
+------------
+
+Find SwiftOverlay, deferring to the associated <Overlay>Config.cmake when requested.
+This is meant to find the platform overlays to be linked by the Supplemental libraries.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+The following :prop_tgt:`IMPORTED` TARGETS may be defined:
+
+ ``SwiftOverlay``
+
+Hint Variables
+^^^^^^^^^^^^^^
+
+ ``SDKROOT`` (environment variable)
+   Set the path to the Swift SDK Root.
+   This only affects Windows and Android builds.
+
+ ``Swift_SDKROOT``
+   Set the path to the Swift SDK installation.
+   This affects Linux, Android, and Windows builds.
+   Apple builds always use the overlay provided by the SDK.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+The module may set the following variables if `swiftDarwin_DIR` is not set and targeting the swiftDarwin overlay.
+
+ ``swiftDarwin_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftDarwin_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftGlibc_DIR` is not set and targeting the GlibC overlay.
+
+ ``swiftGlibc_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftGlibc_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftWinSDK_DIR` is not set and targeting the WinSDK overlay.
+
+ ``swiftWinSDK_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftWinSDK_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftCRT_DIR` is not set and targeting the CRT overlay.
+
+ ``swiftCRT_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftCRT_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+The module may set the following variables if `swiftAndroid_DIR` is not set and targeting the Android overlay.
+
+ ``swiftAndroid_DIR_FOUND``
+   true if the overlay was found
+
+ ``swiftAndroid_INCLUDE_DIR``
+   the directory containing the corresponding overlay swift module folder
+
+
+#]=======================================================================]
+
+include_guard(GLOBAL)
+
+include(FindPackageHandleStandardArgs)
+include(PlatformInfo)
+
+# This was loosely modelled after other find modules
+# (namely FindGLEW), where the equivalent parameter
+# is not stored in cache (possibly because we want
+# the project importing it to be able to
+# it "immediately")
+if(NOT DEFINED SwiftOverlay_USE_STATIC_LIBS)
+  set(SwiftOverlay_USE_STATIC_LIBS OFF)
+  if(NOT BUILD_SHARED_LIBS AND NOT APPLE)
+    set(SwiftOverlay_USE_STATIC_LIBS ON)
+  endif()
+endif()
+
+if(SwiftOverlay_DIR)
+  find_package(SwiftOverlay CONFIG)
```

I think it still makes sense to propagate of the `QUIET` and `REQUIRED` keywords like [find_dependency](https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html#module:CMakeFindDependencyMacro) does (since `find_package` does not do that, at least from a quick skim of the source code)

```suggestion
  if(${CMAKE_FIND_PACKAGE_NAME}_FIND_REQUIRED)
    list(APPEND args REQUIRED)
  endif()
  if(${CMAKE_FIND_PACKAGE_NAME}_FIND_QUIETLY)
    list(APPEND args QUIET)
  endif()
  find_package(SwiftOverlay CONFIG ${args})
```